### PR TITLE
chore(proxy): newtype for Oid

### PR DESCRIPTION
**Pull request by xla**
To make the handling of object ids more safe, provide conversions and
serializations to callers, we introduce a newtype around the git2::Oid.

Preparation for #602

### CODE REVIEW

File: proxy/coco/src/oid.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+use radicle_surf::vcs::git::git2;
+
+/// Unique identity of any object (commit, tree, blob, tag).
+#[derive(Clone, Debug, PartialEq)]
+pub struct Oid(git2::Oid);
+
+impl Oid {
+    /// Consume to get the inner [`git2::Oid`].
+    pub fn inner(self) -> git2::Oid {
```
Comment (kim) [inline proxy/coco/src/oid.rs:17]: You can actually just do `pub struct Oid { pub inner: git2::Oid }`.

File: proxy/coco/src/source.rs [hunk 2]
```diff
@@ -427,7 +429,7 @@ where
 
     let last_commit = browser
         .last_commit(commit_path)?
-        .map(|c| CommitHeader::from(&c));
+        .map(|c| CommitHeader::try_from(&c).expect("unable to convert to commit header"));
```
Comment (kim) [inline proxy/coco/src/source.rs:432]: :scream_cat:

File: proxy/coco/src/oid.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+use std::convert::TryFrom;
+use std::fmt;
+use std::str;
+
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+use radicle_surf::vcs::git::git2;
+
+/// Unique identity of any object (commit, tree, blob, tag).
+#[derive(Clone, Debug, PartialEq)]
```
Comment (kim) [inline proxy/coco/src/oid.rs:12]: `git2::Oid` is `Copy`, which you should probably derive here, too

**Review by kim — APPROVED** [review]
lgtm

File: proxy/coco/src/source.rs (hunk 2 above)
Comment (xla) [reply proxy/coco/src/source.rs]: Turns out it wasn't necessary.

File: proxy/coco/src/oid.rs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+// Serialization.
+
+impl<'de> Deserialize<'de> for Oid {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Self::try_from(s.as_ref()).map_err(de::Error::custom)
```
[the remark this answers is not in the document]
Comment (xla) [reply proxy/coco/src/oid.rs]: :black_heart: :black_heart: :black_heart:

File: proxy/api/src/http/source.rs [hunk 5]
```diff
@@ -680,7 +682,8 @@ mod test {
         let urn = platinum_project.urn();
 
         let branch = git::Branch::local("master");
-        let head = "223aaf87d6ea62eef0014857640fd7c8dd0f80b5";
+        let head = coco::oid::Oid::try_from("223aaf87d6ea62eef0014857640fd7c8dd0f80b5")
```
Comment (cloudhead) [inline proxy/api/src/http/source.rs:685]: It's amazing that `coco::oid::Oid` only uses 4 distinct letters.

**Review by cloudhead — APPROVED** [review]
Good change

File: proxy/api/src/http/source.rs (hunk 5 above)
Comment (xla) [reply proxy/api/src/http/source.rs:685]: By desing, lol.

Comment (kim) [reply proxy/api/src/http/source.rs:685]: lol. It's pronounced "cocoid".

**Pull request merged**